### DISCUSSION TITLE
is it possible to read 2 D array as single vector

### DISCUSSION THREAD

**Discussion by dhirajnitk** [nlohmann/json#2555] [Q&A] [answered]
I am working on 2d vectors but  they are not contiguous./ I am working on   opencv Matrix , I  dont wish to point single row to 
vector data of that individual vector.

STL Vector data storage is on heap. so is there any mean to fetch entire 2 d vector as one d vector?

**Comment by nlohmann** [comment, marked as answer]
I do not understand. What do you want the library to do? What did you try?

**Reply by dhirajnitk** [reply]
Rather than receive vector<vector<T>> to deserialize [[1,2,3],[4,5,6]],  I would prefer  the library to have an option to fetch contiguous data in vector<T> & return Rows/Columns(other can be calculated from size of 1 D vector i.e something like this  `pair<int,vector<T>> `

**Reply by nlohmann** [reply]
For this, you would need to define your own SAX handler, see https://json.nlohmann.me/features/parsing/sax_interface/.
